engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -1773,6 +1773,140 @@ static int SetSound(lua_State* L) {
     return 0;
 }
 
+/*# create a buffer resource
+ * This function creates a new buffer resource that can be used in the same way as any buffer created during build time.
+ * The function requires a valid buffer created from either [ref:buffer.create] or another pre-existing buffer resource.
+ * If the buffer has been created by a script, the new resource will take ownership of the buffer, meaning the buffer
+ * will not automatically be removed when the lua reference to the buffer is garbage collected. This behaviour can
+ * be overruled by specifying 'transfer_ownership = false' in the argument table.
+ *
+ * @name resource.create_buffer
+ *
+ * @param path [type:string] The path to the resource.
+ * @param table [type:table] A table containing info about how to create the buffer. Supported entries:
+ *
+ * * `buffer`
+ * : [type:buffer] the buffer resource to bind to this resource
+ *
+ * * `transfer_ownership`
+ * : [type:boolean] optional flag to determine wether or not the resource should take over ownership of the buffer object (default true)
+ *
+ * @return path [type:hash] Returns the buffer resource path
+ *
+ * @examples
+ * Create a buffer object and bind it to a buffer resource
+ *
+ * ```lua
+ * function init(self)
+ *     local size = 1
+ *     local positions = {
+ *         -- triangle 1
+ *          size,  size, 0,
+ *         -size, -size, 0,
+ *          size, -size, 0,
+ *         -- triangle 2
+ *          size, size,  0,
+ *         -size,  size, 0,
+ *         -size, -size, 0,
+ *     }
+ *
+ *     local buffer_handle = buffer.create(#positions, {
+ *         {
+ *             name  = hash("position"),
+ *             type  = buffer.VALUE_TYPE_FLOAT32,
+ *             count = 3
+ *         }
+ *     })
+ *
+ *     local stream = buffer.get_stream(buffer_handle, hash("position"))
+ *
+ *     -- transfer vertex data to buffer
+ *     for k=1,#positions do
+ *         stream[k] = positions[k]
+ *     end
+ *
+ *     local my_buffer = resource.create_buffer("/my_buffer.bufferc", { buffer = buffer_handle })
+ *     go.set("/go#mesh", "vertices", my_buffer)
+ * end
+ * ```
+ */
+static int CreateBuffer(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+
+    const char* path_str     = luaL_checkstring(L, 1);
+    const char* resource_ext = ".bufferc";
+
+    dmhash_t canonical_path_hash = 0;
+    PreCreateResource(L, path_str, resource_ext, &canonical_path_hash);
+
+    luaL_checktype(L, 2, LUA_TTABLE);
+    lua_pushvalue(L, 2);
+
+    lua_getfield(L, -1, "buffer");
+    dmScript::LuaHBuffer* lua_buffer = dmScript::CheckBuffer(L, -1);
+    lua_pop(L, 1); // "buffer"
+
+    bool transfer_ownership = CheckFieldValue<bool>(L, -1, "transfer_ownership", true);
```

Comment:
Q: what do we want the default to be for this? I would think that the most common use case will be to create the buffer in a script and bind it to a resource that can be used in various places, where you probably want to always transfer the ownership. But maybe I'm wrong?